tools/cross_runner/kernels/conv_nchw_fp16.cpp

```diff
@@ -290,13 +303,22 @@ extern "C" _GENX_MAIN_ void convolution_nchw_nondpas(
 	// but if data types are different then this cast accumulator to output type
     //vector<DT_OUT, ACCU_REG_SIZE> output_row_0 = vector<DT_OUT, ACCU_REG_SIZE>(accu_row_0);
     matrix<DT_OUT, BLOCK_BATCH, ACCU_REG_SIZE> output_row_0 = matrix<DT_OUT, BLOCK_BATCH, ACCU_REG_SIZE>(accu_row_0);
-	
+
 	const uint output_batch_offset = batch_id * BLOCK_BATCH * output_height * output_width * output_channels;
-    const uint output_oc_chunk_offset = oc_chunk_id * BLOCK_OC * output_height * output_width;
-    const uint output_w_chunk_offset = w_chunk_id * BLOCK_W;
-    const uint output_h_chunk_offset = h_chunk_id * BLOCK_H * output_width;
-    uint32_t output_offset = (output_batch_offset + output_oc_chunk_offset + output_h_chunk_offset + output_w_chunk_offset) * sizeof(DT_OUT);
-	
-	store_output_wc8_as_nchw(surface_output, output_row_0.row(0), output_width, output_height, output_offset, w_chunk_id);
-	//store_output_wc8_as_nchw(surface_output, output_row_0.row(1), output_width, output_height, output_offset + output_height * output_width * output_channels * sizeof(DT_OUT), w_chunk_id);
-}
+#if OUTPUT_LAYOUT_IS_NHWC
```

Comment:
What is performance impact if this is runtime condition vs compile time condition?
Could you collect some perf data?

Reply:
The runtime condition is faster, so I'll make that change